domain/src/main/java/com/thoughtworks/go/domain/materials/git/builder/ToRevisionBuilder.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.domain.materials.git.builder;
+
+import com.thoughtworks.go.util.command.CommandArgument;
+import com.thoughtworks.go.util.command.CommandLine;
+
+public class ToRevisionBuilder {
+    private final Builder mainBuilder;
+    private CommandLine git;
+
+    ToRevisionBuilder(GitLogCommandBuilder mainBuilder, CommandLine git) {
+        this.git = git;
+        this.mainBuilder = mainBuilder;
+    }
+
+    public Builder toRevision(String toRevision) {
+        CommandArgument fromRevisionArg = this.git.getArguments().remove(this.git.getArguments().size() - 1);
+        this.git.withArg(String.format("%s..%s", fromRevisionArg.forCommandLine(), toRevision));
+        return mainBuilder;
+    }
+}
```

Comment:
What's the reason behind the three Builders? `ToRevisionBuilder` doesn't implement `Builder` and also seems to mutate the arguments.

Also, if `fromRevision` and `toRevision` are always coupled, it might make sense to just have a function with two arguments.

Reply:
@akshaydewan - No reason. Initially, I thought there will be multiple combinations with the revision. But in the end, I realized have only two scenarios. I will make a change.

Reply:
Done

Reply:
Do we still need `WithBranchBuilder`?

Reply:
Yes, It is to build command in proper order - `git log -1 branch` or `git log -1 branch --pretty=...` and not `git log branch -1`